homeassistant/components/rointe/sensor.py

```diff
@@ -0,0 +1,160 @@
+"""A sensor for the current Rointe radiator temperature."""
+from __future__ import annotations
+
+from datetime import datetime
+
+from rointesdk.device import RointeDevice
+
+from homeassistant.components.sensor import (
+    SensorDeviceClass,
+    SensorEntity,
+    SensorEntityDescription,
+    SensorStateClass,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import ENERGY_KILO_WATT_HOUR, POWER_WATT, TEMP_CELSIUS
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity import EntityCategory
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.typing import StateType
+
+from .const import DOMAIN, ROINTE_COORDINATOR
+from .coordinator import RointeDataUpdateCoordinator
+from .rointe_entity import RointeRadiatorEntity
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up the radiator sensors from the config entry."""
+    coordinator: RointeDataUpdateCoordinator = hass.data[DOMAIN][entry.entry_id][
+        ROINTE_COORDINATOR
+    ]
+
+    # Hook a callback for discovered entities for the sensor entities.
+    coordinator.add_entities_for_seen_keys(
+        async_add_entities,
+        [RointeHaSensor, RointeEnergySensor, RointePowerSensor],
+        "sensor",
+    )
+
+
+class RointeHaSensor(RointeRadiatorEntity, SensorEntity):
+    """Sensor for the radiator current temperature."""
+
+    def __init__(
+        self,
+        radiator: RointeDevice,
+        coordinator: RointeDataUpdateCoordinator,
+    ) -> None:
+        """Init the sensor entity."""
+
+        description = SensorEntityDescription(
+            key="current_temperature",
+            name="Current Temperature",
+            native_unit_of_measurement=TEMP_CELSIUS,
+            device_class=SensorDeviceClass.TEMPERATURE,
+            state_class=SensorStateClass.MEASUREMENT,
+            entity_category=EntityCategory.DIAGNOSTIC,
```

Comment:
Is this the right category? This sensor's only purpose is diagnostics info? Usually, current temperature would just be a "normal" entity.